enterprise/tailnet/coordinator.go

```diff
@@ -0,0 +1,507 @@
+package tailnet
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/coderd/database"
+	agpl "github.com/coder/coder/tailnet"
+)
+
+// NewCoordinator creates a new high availability coordinator
+// that uses PostgreSQL pubsub to exchange handshakes.
+func NewCoordinator(logger slog.Logger, pubsub database.Pubsub) (agpl.Coordinator, error) {
```

Comment:
I'm not opposed, but it kinda makes me feel like I'm dialing something. `New` is sucky though, because it doesn't give indication that it's actually creating something that runs, not just allocates a new chunk of mem.